Makefile

```diff
@@ -235,19 +226,19 @@ CLEAN_TARGETS += clean-vendor
 clean-vendor:
 	-rm -rf $(VENDOR_DIR)
 
-CLEAN_TARGETS += clean-glide-cache
-.PHONY: clean-glide-cache
-## Removes the ./glide directory.
-clean-glide-cache:
-	-rm -rf ./.glide
-
-$(VENDOR_DIR): glide.lock glide.yaml
-	$(GLIDE_BIN) --quiet install --strip-vendor
-	touch $(VENDOR_DIR)
-
 .PHONY: deps
 ## Download build dependencies.
-deps: $(VENDOR_DIR)
+deps: $(DEP_BIN) $(VENDOR_DIR)
+
+# install dep (see https://golang.github.io/dep/docs/installation.html)
+$(DEP_BIN):
+	@echo "Installing 'dep' in $(GOPATH)/bin"
+	@mkdir -p $(GOPATH)/bin
+	curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```

Comment:
Can we stick that to a specific version please? I'm against using master for such a crucial thing.

The [install script](https://raw.githubusercontent.com/golang/dep/master/install.sh) that one can set the environment `DEP_RELEASE_TAG`. I suggest to use `v0.4.1` because that [seems to be the latest release](https://github.com/golang/dep/releases). We should not only check for dep but also for its version and update dep if needed be.

To not overwrite the global version of `deb` we should install it in `tmp/bin` by setting the `INSTALL_DIRECTORY` environment variable. Then we're completely autonomous and don't rely on a global dep or glide.

Reply:
That's a very fair point. Let me see what I can do about it :)

Reply:
I don't even thing we need the whole install script piping to sh. If all there's to it is to download `wget https://github.com/golang/dep/releases/download/v0.4.1/dep-linux-amd64` or some other arch. Also we should check the SHA256 hash.